bdk_chain/src/keychain.rs

```diff
@@ -91,3 +91,56 @@ impl<K, P> ForEachTxout for KeychainChangeSet<K, P> {
         self.chain_graph.for_each_txout(f)
     }
 }
+
+/// Balance differentiated in various categories
+#[derive(Debug, PartialEq, Eq, Clone, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize, serde::Serialize),
+    serde(crate = "serde_crate",)
+)]
+pub struct Balance {
+    /// All coinbase outputs not yet matured
+    pub immature: u64,
+    /// Unconfirmed UTXOs generated by a wallet tx
+    pub trusted_pending: u64,
+    /// Unconfirmed UTXOs received from an external wallet
+    pub untrusted_pending: u64,
+    /// Confirmed and immediately spendable balance
+    pub confirmed: u64,
+}
+
+impl Balance {
+    /// Get sum of trusted_pending and confirmed coins
+    pub fn get_spendable(&self) -> u64 {
+        self.confirmed + self.trusted_pending
+    }
+
+    /// Get the whole balance visible to the wallet
+    pub fn get_total(&self) -> u64 {
+        self.confirmed + self.trusted_pending + self.untrusted_pending + self.immature
+    }
```

Comment:
These two methods were copy pasted from BDK. I am not sure about this though. Why is `trusted_pending` more "spendable" than `untrusted_pending`? I'd opt to remove `get_spendable` and rename `get_total` to `total`.

cc @afilini

Reply:
I did the above ^